sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -308,11 +307,18 @@ public static HttpHeaders createHttpHeadersFromClientOptions(ClientOptions clien
             return null;
         }
 
-        List<HttpHeader> httpHeaderList = new ArrayList<>();
-        clientOptions.getHeaders().forEach(
-            header -> httpHeaderList.add(new HttpHeader(header.getName(), header.getValue())));
+        Iterator<Header> headerIterator = clientOptions.getHeaders().iterator();
+        if (!headerIterator.hasNext()) {
+            return null;
+        }
+
+        HttpHeaders headers = new HttpHeaders();
+        do {
+            Header header = headerIterator.next();
+            headers.set(header.getName(), header.getValue());
+        } while (headerIterator.hasNext());
 
-        return httpHeaderList.isEmpty() ? null : new HttpHeaders(httpHeaderList);
+        return headers;
```

Comment:
I was always a little bothered by this code. Is it possible that we can add a `HttpHeaders.add(Header)` and / or `HttpHeaders.set(Header)` methods, and then reuse the existing `Header` from the ClientOptions, rather than unpacking and repacking into a copy of the header? Are we being needlessly defensive here?

Reply:
This is a good question and I don't have a good grasp on

1. How often is this being used
2. When this is being used are the passed headers treated as immutable by what passed them

This isn't a common hot path so I'd rather punt this to an issue so we can get another round of performance numbers with the changes in this PR.